apps/src/puzzleRatingUtils.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * @overview A helper class for all actions associated with Puzzle
+ * Ratings, aka the Fun-O-Meter.
+ */
+
+var PuzzleRatingUtils = {};
+
+module.exports = PuzzleRatingUtils;
+
+var dom = require('./dom');
+
+/**
+ * Construct the puzzle rating buttons themselves
+ *
+ * @returns {Element} div containing puzzle ratng buttons with attached
+ *          click handlers
+ */
+PuzzleRatingUtils.buildPuzzleRatingButtons = function () {
+  var buttonContainer = document.createElement('div');
+  buttonContainer.id = 'puzzleRatingButtons';
+  buttonContainer.innerHTML = require('./templates/puzzleRating.html.ejs')();
+
+  var buttons = buttonContainer.querySelectorAll('.puzzle-rating-btn');
+  var buttonClickHandler = function () {
+    for (var i = 0, button; (button = buttons[i]); i++) {
+      if (button != this) {
+        $(button).removeClass('enabled');
+      }
+    }
+    $(this).toggleClass('enabled');
+  };
+  for (var i = 0, button; (button = buttons[i]); i++) {
+    dom.addClickTouchEvent(button, buttonClickHandler);
+  }
+
+  return buttonContainer;
+};
+
+/**
+ * @typedef {Object} PuzzleRating
+ *
+ * @property {number} script_id
+ * @property {number} level_id
+ * @property {number|string} rating - can be a number or an
+ *           integer-parseable string
+ */
+
+/**
+ * Private getter/localStorage proxy
+ * @returns {PuzzleRating[]} - ratings
+ */
+PuzzleRatingUtils.getPuzzleRatings_ = function () {
+  var ratings = localStorage.getItem('puzzleRatings');
+  try {
+    return ratings ? JSON.parse(ratings) : [];
+  } catch (e) {
+    return [];
+  }
+};
+
+/**
+ * Private setter/localStorage proxy
+ * @param {PuzzleRating[]} ratings
+ */
+PuzzleRatingUtils.setPuzzleRatings_ = function (ratings) {
+  localStorage.setItem('puzzleRatings', JSON.stringify(ratings));
+};
+
+/**
+ * Private deleter/localStorage proxy
+ * @param {PuzzleRating} rating
+ */
+PuzzleRatingUtils.removePuzzleRating_ = function (rating) {
+  var ratings = PuzzleRatingUtils.getPuzzleRatings_().filter(function (other) {
+    var otherEqualsRating = (rating.level_id == other.level_id &&
+      rating.script_id == other.script_id &&
+      rating.rating == other.rating);
+    return !otherEqualsRating;
+  });
+  PuzzleRatingUtils.setPuzzleRatings_(ratings);
+};
+
+/**
+ * Cache the selected rating, to be submitted (possibly with other
+ * cached ratings) at some later point
+ *
+ * @param {jQuery} container - some element that contains the buttons
+ * @param {Object} options - other data to be submitted along with the
+ *        rating. Usually script_id and level_id
+ */
+PuzzleRatingUtils.cachePuzzleRating = function (container, options) {
+  var selectedButton = container.querySelector('.puzzle-rating-btn.enabled');
+  if (selectedButton) {
+    options.rating = selectedButton.getAttribute('data-value');
+    var ratings = PuzzleRatingUtils.getPuzzleRatings_();
+    ratings.push(options);
+    PuzzleRatingUtils.setPuzzleRatings_(ratings);
+  }
+};
```

Comment:
Right now we pass in an options object, mutate it, and push it to a list. As the code exists right now, the only caller passes in an object literal, so we don't need to worry about it being mutated elsewhere, but I think we could make this a little cleaner in one of two ways:
(1) pass in scriptId, levelId, and then construct our options object inside cachePuzzleRating
(2) in cachePuzzleRating do `$.extend({}, options, { rating: selectedButton.getAttribute('data-value') }) to make it explicit that we have a local object that cant be modified externally